### DISCUSSION TITLE
How to mutate data in ManageRelatedRecords form function?

### DISCUSSION THREAD

**Discussion by JilsonSaji** [filamentphp/filament#11710] [Help]
### Package

Panel builder

### Package Version

v3.0.0

### How can we help you?

I have created a OneToMany relationship and I am using the Filament\Resources\Pages\ManageRelatedRecords for creating the child element. I need to mutate the form data before saving it to the Db..!
Any suggestions????

**Comment by leandrocfe** [comment]
https://filamentphp.com/docs/3.x/actions/prebuilt-actions/create#customizing-data-before-saving
?

**Reply by JilsonSaji** [reply]
This function doesn't seem to be applied to the ManageRelatedRecords ...!

**Comment by zeeshantariq08** [comment]
you can use this function in CreatePage.php
```
 protected function mutateFormDataBeforeCreate(array $data): array
    {

        $data['created_by'] = auth()->id();
        return $data;
    }
```

**Reply by JilsonSaji** [reply]
Doesn't seems to work

**Reply by leandrocfe** [reply]
Try it in your ManagePage

```php
Tables\Actions\CreateAction::make()
    ->mutateFormDataUsing(function (array $data): array {

        //mutate form data
        $data['user_id'] = auth()->id();
        
        return $data;
    })
```
